### ISSUE TITLE
Render different types of rail stations differently

### ISSUE THREAD

**Issue by wmisener** [ZeLonewolf/openstreetmap-americana#860]
As discussed in #790 and #845, light and heavy rail are viewed as distinct modes in the US, and it would be desirable if the project treated light rail stations distinctly from subway and heavy rail stations. I suggested this basic three way division in https://github.com/ZeLonewolf/openstreetmap-americana/issues/790#issuecomment-1431917320. There are also more exotic rail types, such as funiculars and monorails, that could also get icons, and are tagged distinctly in OSM with `station=*`. However, such rich rendering is currently blocked by Open Map Tiles, as the tiles only currently have two categories for rail stations: subways and everything else. Sensibly in light of this limitation, #845 implemented a single icon for all rail stations, with rail stations coming in at z12, with subway stations following at z14. This leads to the unfortunate side effect that light rail, and even funicular, stations render two zoom levels before subway stations, when they are really of similar or lesser importance.

Example: Los Angeles at z12:
<img width="605" alt="Screen Shot 2023-04-12 at 10 09 10 PM" src="https://user-images.githubusercontent.com/58491489/231659538-cc697b0c-aafe-4cba-86c1-38d2ff0ca263.png">
Stations along the A and E light rail lines (all tagged `station=light_rail`, e.g. Expo/Vermont) and the Angels Flight (`station=funicular`, e.g. Olive Street) render at z12, but the B and D line subway stations (e.g. Westlake/MacArthur Park, tagged`station=subway`) do not.

This problem has already been discussed, but I'm opening a separate issue to make sure we track it. This would be unblocked by https://github.com/openmaptiles/openmaptiles/issues/1517.
